src/solver/infeasible-problem-analysis/constraint-slack-analysis.cpp

```diff
@@ -53,9 +59,22 @@ void ConstraintSlackAnalysis::run(MPSolver* problem)
     }
 
     hasDetectedInfeasibilityCause_ = true;
+
+    sortSlackVariablesByValue();
+    trimSlackVariables();
 }
 
-void ConstraintSlackAnalysis::addSlackVariables(MPSolver* problem)
+void ConstraintSlackAnalysis::selectConstraintsToWatch(MPSolver* problem)
+{
+    ConstraintsFactory factory;
+    std::regex rgx = factory.constraintsFilter();
+    std::copy_if(problem->constraints().begin(),
+                 problem->constraints().end(),
```

Comment:
```suggestion
    std::ranges::copy_if(problem->constraints(),
```
https://en.cppreference.com/w/cpp/algorithm/ranges/copy